approov-service/src/main/java/io/approov/service/okhttp/ApproovServiceMutator.java

```diff
@@ -0,0 +1,311 @@
+     *                          the cause of the failure.
+     */
+    default void handlePrecheckResult(Approov.TokenFetchResult approovResults) throws ApproovException {
+        Approov.TokenFetchStatus status = approovResults.getStatus();
+        String arc = approovResults.getARC();
+        String rejectionReasons = approovResults.getRejectionReasons();
+        switch (status) {
+            case REJECTED:
+                throw new ApproovRejectionException("precheck: " + status.toString() + ": " + arc + " " + rejectionReasons, arc, rejectionReasons);
+            case NO_NETWORK:
+            case POOR_NETWORK:
+            case MITM_DETECTED:
+                throw new ApproovNetworkException("precheck: " + status.toString());
+            case SUCCESS:
+            case UNKNOWN_KEY:
+                break;
+            default:
+                throw new ApproovException("precheck:" + status.toString());
+        }
+    }
+
+    /**
+     * Decides how to handle the token fetch result from an
+     * ApproovService.fetchToken() operation.
+     *
+     * @param approovResults the TokenFetchResult obtained by
+     *                       ApproovService.fetchToken()
+     * @throws ApproovException The implementation can either return, taking no
+     *                          action or throw an ApproovException encoding
+     *                          the cause of the failure.
+     */
+    default void handleFetchTokenResult(Approov.TokenFetchResult approovResults) throws ApproovException {
+        Approov.TokenFetchStatus status = approovResults.getStatus();
+        switch (status) {
+            case NO_NETWORK:
+            case POOR_NETWORK:
+            case MITM_DETECTED:
+                throw new ApproovNetworkException("fetchToken: " + status.toString());
+            case SUCCESS:
+                break;
+            default:
+                throw new ApproovException("fetchToken: " + status.toString());
+        }
+    }
+
+    /**
+     * Decides how to handle the token fetch result from an
+     * ApproovService.fetchSecureString() operation.
+     *
+     * @param approovResults the TokenFetchResult obtained by
+     *                       ApproovService.fetchSecureString()
+     * @param operation the operation type ("lookup" or "definition"); "lookup"
+     *                  indicates that an existing value was requesterd, while
+     *                  "definition" indicates that a new value was being added
+     *                  or set
+     * @param key the secure string key
+     * @throws ApproovException The implementation can either return, taking no
+     *                          action or throw an ApproovException encoding
+     *                          the cause of the failure
+     */
+    default void handleSecureStringResult(Approov.TokenFetchResult approovResults, String operation, String key) throws ApproovException {
```

Comment:
I think this should be renamed for consistency. We have:

* `handlePrecheckResult` (handles `.precheck()`)
* `handleFetchTokenResult` (handles `.fetchToken()`)
* `handleSecureStringResult` (handles `.fetchSecureString()`)

Thus, it seems this should be:

```suggestion
    default void handleFetchSecureStringResult(Approov.TokenFetchResult approovResults, String operation, String key) throws ApproovException {
```